schemas/manifest.fbs

```diff
@@ -5,11 +5,22 @@ table Uuid {
    low: uint64;
 }
 
+table DbParent {
+   // Optional path to a parent database from which this database was cloned
+   parent_path: string (required);
+
+   // Optional parent checkpoint ID
+   parent_checkpoint: Uuid (required);
+}
+
 // Manifest to persist state of DB.
 table ManifestV1 {
     // Id of manifest. Manifest file name will be derived from this id.
     manifest_id: ulong;
 
+    // Optional details about the parent checkpoint for the database
+    parent: DbParent;
```

Comment:
🤔 Do we need reverse poitnters in the parent's manifest as well?

I'm thinking about case where we clone the DB multiple times (which is what I'm actually shooting for - splitting the database into multiple instances). I assume GC would run for each DB separately and it somehow needs to know when it can GC parent's data files?

Reply:
Each clone maintains its own checkpoint in the parent's manifest, so the parent GC won't delete SSTs out from underneath the clone. The clone GC tracks when the clone no longer references any of the parent SSTs, and when that happens it deletes its checkpoint from the parent manifest.

Reply:
Nice, that's pretty smart. Thanks for the context!